apps/src/code-studio/components/progress/BulkLessonVisibilityToggle.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import {getStore} from '../../../redux';
+import PropTypes from 'prop-types';
+import Button from '@cdo/apps/templates/Button';
+import FontAwesome from '../../../templates/FontAwesome';
+import ReactTooltip from 'react-tooltip';
+import {toggleHiddenLesson} from '@cdo/apps/code-studio/hiddenLessonRedux';
+import {unitCalendarLesson} from '../../../templates/progress/unitCalendarLessonShapes';
+import _ from 'lodash';
+import style from './bulk-lesson-visibility-toggle.module.scss';
+
+function BulkLessonVisibilityToggle({lessons, sectionId, scriptName}) {
+  const tooltipId = _.uniqueId();
+
+  return (
+    <div className={style.container}>
+      <Button
+        text="Show All Lessons"
+        icon="eye"
+        color={Button.ButtonColor.gray}
+        onClick={() =>
+          toggleHiddenLessons(scriptName, sectionId, lessons, false)
+        }
+      />
+      <Button
+        text="Hide All Lessons"
+        icon="eye-slash"
+        color={Button.ButtonColor.gray}
+        onClick={() =>
+          toggleHiddenLessons(scriptName, sectionId, lessons, true)
+        }
+      />
+      <span data-tip data-for={tooltipId}>
+        <FontAwesome icon="info-circle" className={style.infoTipIcon} />
+      </span>
+      <ReactTooltip id={tooltipId} effect="solid">
+        <p>
+          Make all lessons in this Unit visible or hidden for your students.
+        </p>
+      </ReactTooltip>
+    </div>
+  );
+}
+
+BulkLessonVisibilityToggle.propTypes = {
+  lessons: PropTypes.arrayOf(unitCalendarLesson),
+
+  // redux provided
+  sectionId: PropTypes.number,
+  scriptName: PropTypes.string.isRequired
+};
+
+function toggleHiddenLessons(scriptName, sectionId, lessons, hidden) {
+  lessons.forEach(lesson => {
+    // For some reason, sectionId is a number here, and needs to be a string
+    // for the redux toggle stuff to work.
```

Comment:
is this related to the propTypes above? https://github.com/code-dot-org/code-dot-org/pull/48863/files#diff-df37c341b4e24afa4446ab42d92a84705737c3bc034cf85dfdcd51183bedc078R50. or is there something else I'm missing?